hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java

```diff
@@ -686,4 +696,17 @@ void notifyGroupRemove(RaftGroupId gid) {
   void notifyGroupAdd(RaftGroupId gid) {
     raftGids.add(gid);
   }
+
+  void handleLeaderChangedNotification(RaftGroupMemberId groupMemberId,
+                                       RaftPeerId raftPeerId) {
+    LOG.info("Leader change notification received for group: {} with new " +
+        "leaderId: {}", groupMemberId.getGroupId(), raftPeerId);
+    // Save the reported leader to be sent with the report to SCM
+    leaderIdMap.put(groupMemberId.getGroupId(), raftPeerId);
+    // Publish new reports with leaderID
+    context.getParent().getReportManager().getReportPublisher(
+        PipelineReportsProto.class).run();
```

Comment:
Instead of modifying ReportManager, you directly add PipelineReport to context
```context.addReport(context.getParent().getContainer().getPipelineReport())```